runtime/queries/column_timeseries.go

```diff
@@ -103,24 +104,30 @@ func (q *ColumnTimeseries) Resolve(ctx context.Context, rt *runtime.Runtime, ins
 		}
 
 		measures := normaliseMeasures(q.Measures, q.Pixels != 0)
-		dateTruncSpecifier := convertToDateTruncSpecifier(timeRange.Interval)
+		timeBucketSpecifier := convertToTimeBucketSpecifier(timeRange.Interval)
 		tsAlias := tempName("_ts_")
 		temporaryTableName := tempName("_timeseries_")
+
+		timezone := "UTC"
+		if q.TimeZone != "" {
+			timezone = q.TimeZone
+		}
+
 		querySQL := `CREATE TEMPORARY TABLE ` + temporaryTableName + ` AS (
 			-- generate a time series column that has the intended range
 			WITH template as (
 			SELECT
 				range as ` + tsAlias + `
 			FROM
 				range(
-				date_trunc('` + dateTruncSpecifier + `', TIMESTAMP '` + timeRange.Start.AsTime().Format(IsoFormat) + `'),
-				date_trunc('` + dateTruncSpecifier + `', TIMESTAMP '` + timeRange.End.AsTime().Format(IsoFormat) + `'),
-				INTERVAL '1 ` + dateTruncSpecifier + `')
+				time_bucket(INTERVAL '` + timeBucketSpecifier + `', TIMESTAMPTZ '` + timeRange.Start.AsTime().Format(IsoFormat) + `', '` + timezone + `'),
+				time_bucket(INTERVAL '` + timeBucketSpecifier + `', TIMESTAMPTZ '` + timeRange.End.AsTime().Format(IsoFormat) + `', '` + timezone + `'),
+				INTERVAL '` + timeBucketSpecifier + `')
 			),
 			-- transform the original data, and optionally sample it.
 			series AS (
 			SELECT
-				date_trunc('` + dateTruncSpecifier + `', ` + safeName(q.TimestampColumnName) + `) as ` + tsAlias + `,` + getExpressionColumnsFromMeasures(measures) + `
+				time_bucket(INTERVAL '` + timeBucketSpecifier + `', ` + safeName(q.TimestampColumnName) + `::TIMESTAMPTZ, '` + timezone + `') as ` + tsAlias + `,` + getExpressionColumnsFromMeasures(measures) + `
```

Comment:
There's still a SQL injection opportunity here. If it's not feasible to pass as args, consider using `time.LoadLocation` to validate the input timezone in Go before running the query (note: need to `import _ "time/tzdata"` for it to work)